spec/unit/facter/udevadm_spec.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'udevadm fact', type: :fact do
+  subject(:fact) { Facter.fact(:udevadm) }
+
+  before { Facter.clear }
+
+  context 'when kernel is linux and udevadm is present' do
+    before do
+      allow(Facter.fact(:kernel)).to receive(:value).and_return(:linux)
```

Comment:
I think the real fact is an uppercased string
```suggestion
      allow(Facter.fact(:kernel)).to receive(:value).and_return('Linux')
```

Reply:
I copied this from the `systemd` fact and it oddly does seem to be working. I will fix that fact as well.

Reply:
Fixed for all facts in this mod.